### PR TITLE
More C++

### PR DESCRIPTION
**Pull request by robertodr**
Addresses the remaining points in #110 I need to write the solution for the randomness example. Notes:
- I have refactored the examples to adhere to C++11 idioms.
- All C++ examples have been reformatted.

### CODE REVIEW

File: content/test-design.md [hunk 1]
```diff
@@ -361,7 +340,7 @@ Also discuss why some are easier to test than others.
 outputs.  For example, start with the below, then think of some
 what extreme cases/boundary cases there might be.  This example shows
 all of the tests as one function, but you might want to make each test
-function more fine-grained and test only one concept.
+function more    fine-grained and test only one concept.
```
Comment (wikfeldt) [inline content/test-design.md:343]:
```suggestion
function more fine-grained and test only one concept.
```

File: content/test-design.md [hunk 2]
```diff
@@ -85,114 +85,93 @@ Also discuss why some are easier to test than others.
...
-      if(n<0)
-        throw "received negative input";
-      if(n==1)
-        return 1;
-      else
-        return factorial(n-1) * n;
+   // 1
+   /* Computes the factorial of n recursively. */
+   constexpr unsigned int factorial(unsigned int n) {
+      return (n <= 1) ? 1 : (n * factorial(n - 1));
```
Comment (wikfeldt) [inline content/test-design.md:91]: nice to have it ternary and recursive, but the python and julia versions deliberately don't do it for simplicity reasons. I guess it doesn't matter much for this example though

**Review by wikfeldt — APPROVED** [review]
super good, thanks

**Comment by bast** [comment]
Thanks for this and thanks also for the reformatting!

File: content/test-design.md (hunk 2 above)
Comment (bast) [reply content/test-design.md:91]: I also find ternaries a bit hard to read and I am avoiding these in my own code but I think it's OK here since it is a relatively simple example.

Comment (robertodr) [reply content/test-design.md:91]: While a non-recursive function would certainly work and would be OK C++, this example shows a recursion that can be evaluated at compile time because of the `constexpr` qualifier. The ternary `if` is necessary in C++11 `constexpr` function can only have one return statement (no loops, not multi-line conditionals)
